### PR TITLE
Obsavo demo timing

### PR DESCRIPTION
**Pull request by jvgomez**
@dominiquehunziker please take a quick look to this. Is mainly adding some comments that were used to analyze the timing and I would like to keep around.

### CODE REVIEW

File: src/rplidar_nodelet.cpp [hunk 1]
```diff
@@ -81,8 +81,11 @@ namespace rplidar_ros {
     {
       if (initialised)
       {
+        //ros::Time start_scan_time = ros::Time::now();
         boost::mutex::scoped_lock lock(mutex_);
         this->read_scan(); // publish inside this
+        //ros::Time end_scan_time = ros::Time::now();
+        //ROS_INFO("Time take by full laserscan: %f", (end_scan_time-start_scan_time).toSec());
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp]: I would fix the grammar

File: src/rplidar_nodelet.cpp [hunk 2]
```diff
@@ -170,7 +174,7 @@ namespace rplidar_ros {
                     int i = 0, j = 0;
                     for( ; i < count; i++ ) {
                         if (nodes[i].distance_q2 != 0) {
-                            float angle = (float)((nodes[i].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)/64.0f);
+                            float angle = (float)((nodes[i].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)*0.015625); // /64.0f);
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp]: this is now a double literal; convert it to a float literal.

File: src/rplidar_nodelet.cpp [hunk 3]
```diff
@@ -191,14 +195,14 @@ namespace rplidar_ros {
                     while (nodes[i--].distance_q2 == 0);
                     end_node = i+1;
 
-                    angle_min = DEG2RAD((float)(nodes[start_node].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)/64.0f);
-                    angle_max = DEG2RAD((float)(nodes[end_node].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)/64.0f);
+                    angle_min = DEG2RAD((float)(nodes[start_node].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)*0.015625); // /64.0f);
+                    angle_max = DEG2RAD((float)(nodes[end_node].angle_q6_checkbit >> RPLIDAR_RESP_MEASUREMENT_ANGLE_SHIFT)*0.015625); // /64.0f);
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp]: these are now a double literals; convert then to a float literals.

File: src/rplidar_nodelet.cpp [hunk 4]
```diff
@@ -222,6 +226,9 @@ namespace rplidar_ros {
           NODELET_WARN_STREAM("RPLidar: she's dead Jim! [timed out waiting for a full 360 scan]");
           initialised = false;
         }
+
+        //ros::Time end_laserscan_time = ros::Time::now();
+        //ROS_INFO("Time take by laserscan transformation: %f", (end_laserscan_time-start_laserscan_time).toSec());
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp]: fix grammar

File: src/rplidar_nodelet.cpp [hunk 5]
```diff
@@ -262,7 +269,7 @@ namespace rplidar_ros {
       bool reverse_data = (!inverted && reversed) || (inverted && !reversed);
       if (!reverse_data) {
           for (size_t i = 0; i < node_count; i++) {
-              float read_value = (float) nodes[i].distance_q2/4.0f/1000;
+              float read_value = (float) nodes[i].distance_q2*0.00025; // /4.0f/1000;
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp]: convert to float literal

File: src/rplidar_nodelet.cpp [hunk 6]
```diff
@@ -271,7 +278,7 @@ namespace rplidar_ros {
           }
       } else {
           for (size_t i = 0; i < node_count; i++) {
-              float read_value = (float)nodes[i].distance_q2/4.0f/1000;
+              float read_value = (float)nodes[i].distance_q2*0.00025; // /4.0f/1000;
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp]: convert to float literal

File: src/rplidar_nodelet.cpp [hunk 7]
```diff
@@ -152,8 +155,9 @@ namespace rplidar_ros {
     start_scan_time = ros::Time::now();
     op_result = drv->grabScanData(nodes, count);
     end_scan_time = ros::Time::now();
-    scan_duration = (end_scan_time - start_scan_time).toSec() * 1e-3;
-
+    scan_duration = (end_scan_time - start_scan_time).toSec();// * 1e-3;
```
Comment (dominiquehunziker) [inline src/rplidar_nodelet.cpp:155]: this changes the units of the scan_duration; was this a bug or did you make a mistake here?

**Review by dominiquehunziker — CHANGES_REQUESTED** [review]

File: src/rplidar_nodelet.cpp (hunk 7 above)
Comment (jvgomez) [reply src/rplidar_nodelet.cpp:155]: When I did the timing, I timed the whole loop into 3 different parts: data acquisition from serial, transformation to ROS LaserScan and publishing, and the total (which should be the sum). Without this change, times will not match. This value is later assigned to scan_time, which is "time between scans [s]". In fact, I am still not sure if this is the correct time, as it does not measure exactly time between scans. But, in any case, I think it was a bug.

**Comment by jvgomez** [comment]
@dominiquehunziker comments addressed. I do not really know how this new Review feature from github works yet, I do not know if I have to do anything else to notify changes or what.

**Review by dominiquehunziker — APPROVED** [review]

**Pull request merged**